### PR TITLE
q2 base

### PR DESCRIPTION
**Pull request by wave49192**

### CODE REVIEW

File: base.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+def base(base_type,num)
+  sum =  0
+  new_num = num.to_s.reverse.split("")
```
Comment (WatcharitNew) [inline base.rb:3]: I think we can omit `("")`
```suggestion
  new_num = num.to_s.reverse.split
```

File: base.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+def base(base_type,num)
```
Comment (WatcharitNew) [inline base.rb:1]:
```suggestion
def base(base_type, num)
```

File: base.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  sum =  0
+  new_num = num.to_s.reverse.split("")
+
+  new_num.each_with_index do |digit, index|
+    digit = digit.to_i  * (base_type ** index)
+    sum+= digit
```
Comment (WatcharitNew) [inline base.rb:7]:
```suggestion
    sum += digit
```

File: base.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+def base(base_type,num)
+  sum =  0
+  new_num = num.to_s.reverse.split("")
+
+  new_num.each_with_index do |digit, index|
+    digit = digit.to_i  * (base_type ** index)
```
Comment (WatcharitNew) [inline base.rb:6]:
```suggestion
    digit = digit.to_i * (base_type ** index)
```

**Review by WatcharitNew — COMMENTED** [review]
As far as I understand question, it seems that it should support other digit base as well. Like add one more base params and change number to that base
Screenshot 2567-08-21 at 15 10 28

**Comment by kan-datawow** [comment]
we're ruby developer, we can do this :trollface: 
```ruby
num.to_s.to_i(base_type.to_i)
```

File: base.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
+def base_converter(from_base,num,to_base)
+  base10 =  num.to_s.to_i(from_base.to_i)
+
+  return '0' if base10 == 0
+  digits = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ'
```
Comment (WatcharitNew) [inline base.rb:5]: May move to outside function as a constant

File: base.rb [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  result = ''
+  while base10 > 0
+    remainder = base10 % to_base
+    result = digits[remainder] + result
+    base10 /= to_base
+  end
+
+  puts result
```
Comment (WatcharitNew) [inline base.rb:13]: This one if not using please remove

**Review by WatcharitNew — APPROVED** [review]

**Pull request merged**